### PR TITLE
Modularizing

### PR DESCRIPTION
**Pull request by hnykda**
Hey. 

I created from this a proper module, removed unused imports and used argparse. Totally fine if you don't want to merge it, just so you know.

@elifiner should probably know about this as well.

### CODE REVIEW

**Comment by finereli** [comment]
Hey @hnykda, thanks for pinging me on this, I had no idea anybody was using that old module let alone forking and extending it. It's awesome! Looks like maybe @masipcat is the closest thing to an official maintainer of pypm at this point.

**Comment by hnykda** [comment]
Cool, thank for the answer. I think I could really nicely leverage this tool in debugging failed processing of some ETL jobs.

**Review by dmanchon — APPROVED** [review]

**Comment by dmanchon** [comment]
@hnykda @elifiner nice to read you!.

Our usecase a large distributed queue with workers processing tasks in paralel, and pydump was one of the ideas that we are testing in order to post mortem debug the failed ones. The reason to fork is to add python3 support. So far with the changes that we made is working quite fine.

We will try again with the changes and merge after that. Of  course any help, request or issue report is more than welcomed :)

File: pypm/__main__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import argparse
+import importlib
+
+from pypm.pypm import debug_dump
+
+
+def parse_args():
+    parser = argparse.ArgumentParser('pypm - post_mortem debugging of python dumps using pypm.save_dump')
+    parser.add_argument('dump', help='Path to the dump file')
+    parser.add_argument('--debugger', default='ipdb', help='Debugger to use (must implement `post_morgem`)')
```
Comment (masipcat) [inline pypm/__main__.py:11]: Could you change `ipdb` with `pdb`? The first one isn't installed by default.

File: example.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import pandas as pd
```
Comment (masipcat) [inline example.py:1]: I know we are using `requests` library on `crash_test.py`, but I would like to suggest to have a dependency-free `example.py`. We could add complex cases (with external dependencies) in a test script. What do you think?

File: example.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+def f():
+
+    adam = 3
+    beta = 4
+    theta = lambda x: x**2
+    delta = range(10)
+    epsilon = t(2)
```
Comment (masipcat) [inline example.py:8]: `t()` is not defined. Is this the desired behavior?

**Review by masipcat — CHANGES_REQUESTED** [review]
Thank you for your changes :-) I've just left a couple of comments.

**Comment by masipcat** [comment]
Hey @hnykda.

I'll modify the requested changes, resolve the conflicts and merge your changes. 

Thank you again for your contribution!

**Comment by hnykda** [comment]
Hey! Sorry, I wasn't able to get back to it earlier... Thanks

**Pull request merged**